### PR TITLE
base - fix subtype

### PR DESCRIPTION
**Pull request by ttodua**

### CODE REVIEW

File: js/base/Exchange.js [hunk 1]
```diff
@@ -2146,11 +2146,11 @@ module.exports = class Exchange {
     handleSubTypeAndParams (methodName, market = undefined, params = {}) {
         let subType = undefined;
         // if set in params, it takes precedence
-        const subTypeInParams = this.safeString2 (params, 'subType', 'subType');
+        const subTypeInParams = this.safeString2 (params, 'subType', 'defaultSubType');
```
Comment (ttodua) [inline js/base/Exchange.js:2149]: just inversed orders , as `default` prefixed should have less priority as explicit keyword itself (imo).

File: js/base/Exchange.js [hunk 2]
```diff
@@ -2146,11 +2146,11 @@ module.exports = class Exchange {
...
         // avoid omitting if it's not present
         if (subTypeInParams !== undefined) {
             subType = subTypeInParams;
-            params = this.omit (params, [ 'defaultSubType', 'subType' ]);
+            params = this.omit (params, [ 'subType', 'defaultSubType' ]);
```
Comment (ttodua) [inline js/base/Exchange.js:2153]: just inversed orders , as `default` prefixed should have less priority as explicit keyword itself (imo). at one day, we might move all these 'default' prefixed options.

File: js/base/Exchange.js [hunk 3]
```diff
@@ -2162,9 +2162,8 @@ module.exports = class Exchange {
             }
             // if it was not defined in market object
             if (subType === undefined) {
-                const exchangeWideValue = this.safeString2 (this.options, 'defaultSubType', 'subType', 'linear');
-                const methodOptions = this.safeValue (this.options, methodName, {});
-                subType = this.safeString2 (methodOptions, 'defaultSubType', 'subType', exchangeWideValue);
+                const values = this.handleOptionAndParams (undefined, methodName, 'subType', 'linear'); // no need to re-test params here
+                subType = values[0];
```
Comment (ttodua) [inline js/base/Exchange.js:2166]: `handleOptionAndParams` has the correct flow of logic, so it's better we used it there. 
(additionally, when we don't need to check `params`, we provide undefined there).

**Comment by ttodua** [comment]
@carlosmiei this also fixes an important bug working with `subType` as users would be facing issues with `defaultSubType` usage.

**Comment by carlosmiei** [comment]
LGTM ✅ 

image

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**